master/internal/rm/kubernetes_resource_manager.go

```diff
@@ -96,6 +105,22 @@ func (k KubernetesResourceManager) ValidateResourcePool(ctx actor.Messenger, nam
 	return nil
 }
 
+// GetResourcePoolAvailability checks the available resources for a given pool.
+func (k KubernetesResourceManager) GetResourcePoolAvailability(
+	ctx actor.Messenger,
+	name string,
+	slots int,
+) ([]command.LaunchWarning, error) {
+	launchWarnings := []command.LaunchWarning{}
+	if slots == 0 {
+		return launchWarnings, nil
+	}
+	if name != "" && name != KubernetesDummyResourcePool {
```

Comment:
do we need to duplicate the `ValidateResourcePool` check for some reason. I would think this should always return `return nil, nil` until we implement this.

Reply:
updated